### PR TITLE
Dependabot for native code

### LINKED ISSUE #460: Dependabot for native dependencies
Configure dependabot to update dependencies defined in:
* test\OpenTelemetry.AutoInstrumentation.Native.Tests\packages.config
* src\OpenTelemetry.AutoInstrumentation.Native\packages.config

### PR DESCRIPTION
**Pull request by Kielek**
Fixes #460
Fixes leftovers from #486  

Changes proposed in this pull request:
* Dependabot for native code
* update dependabot config after renaming projects (I have forgot to attach this file in previous PR)

Tested on my fork
image

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -99,6 +99,12 @@ updates:
       interval: weekly
     open-pull-requests-limit: 20
 
+  - package-ecosystem: nuget
+    directory: /src/OpenTelemetry.AutoInstrumentation.Native
+      day: sunday
```
Comment (Kielek) [inline .github/dependabot.yml:104]: Candidate for separate PR - change the interval from `weekly` to `daily`. 

Benefits: We will reduce number of new PRs in Sunday/Monday morning. It should be easier to manage it.

**Review by lachmatt — APPROVED** [review]

File: .github/dependabot.yml [hunk 2]
```diff
@@ -99,6 +99,12 @@ updates:
       interval: weekly
     open-pull-requests-limit: 20
 
+  - package-ecosystem: nuget
```
Comment (pjanotti) [inline .github/dependabot.yml:102]: Do we need a follow-up issue/PR to cover Linux and mac builds of native?

**Review by pjanotti — APPROVED** [review]
LGTM

**Review by nrcventura — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

File: .github/dependabot.yml (hunk 2 above)
Comment (Kielek) [reply .github/dependabot.yml:102]: I think that we should treat native code dependencies PR created as a reminder that we should check it also for linux.

BTW based on what I see we have different versions  for Windows and Linux/MacOs.
See https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/d486e2e5c590d2b5314fc326551e0758d0b61eec/src/OpenTelemetry.AutoInstrumentation.Native/CMakeLists.txt#L105
and 
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/d486e2e5c590d2b5314fc326551e0758d0b61eec/src/OpenTelemetry.AutoInstrumentation.Native/packages.config#L3

**Pull request merged**